### PR TITLE
Fix CLI function variables deploying

### PR DESCRIPTION
**Pull request by alibekissakul**
## What does this PR do?

Fixing variable deployment when creating functions

## Test Plan

[x] Manual testing

## Related PRs and Issues

Closes https://github.com/appwrite/sdk-for-cli/issues/59

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: templates/cli/lib/commands/deploy.js.twig
Comment (Meldiron) [inline templates/cli/lib/commands/deploy.js.twig:191]: We should keep this logic here, it's for updating existing function.

File: templates/cli/lib/commands/deploy.js.twig
Comment (Meldiron) [inline templates/cli/lib/commands/deploy.js.twig:227]: This is logic for initial creation. We can remove a lof of code - we don't need to list, ask user if he is sure, and delete..

We could simply run this here:

```php
if(func.variables) {
            // Deploy local variables
            await Promise.all(Object.keys(func.variables).map(async localVariableKey => {
                await functionsCreateVariable({
                    functionId: func['$id'],
                    key: localVariableKey,
                    value: func.variables[localVariableKey],
                    parseOutput: false
                });
            }));
}
```

**Review by Meldiron — CHANGES_REQUESTED** [review]
Thanks for the PR 😇 I left a review.

File: templates/cli/lib/commands/deploy.js.twig
Comment (Meldiron) [inline templates/cli/lib/commands/deploy.js.twig:257]: Amazing, this now looks great 🥳

File: templates/cli/lib/commands/deploy.js.twig
Comment (Meldiron) [inline templates/cli/lib/commands/deploy.js.twig:220]: Sorry I was probably confusing in my last comment.
The logic here is for when a function already exists, and it does:
1. Ask if you are okay loosing variables
2. Delete existing varuables
3. Create variables from local config

So this part has to stay here.

Comment (Meldiron) [reply templates/cli/lib/commands/deploy.js.twig:220]: The addition you made inside catch() is now correct by the way. That looks great 😇

Comment (yaakov123) [reply templates/cli/lib/commands/deploy.js.twig:220]: Why should we delete existing variables. Surely we should only delete variables that are conflicting. That way if there are some variables that exist in the console that were created manually, we won't loose them.

Comment (abnegate) [reply templates/cli/lib/commands/deploy.js.twig:220]: The problem with that is that if you then tried to deploy the function to a different Appwrite instance, the function would be depending on variables that don't exist in the configuration

**Comment by ChandanChainani** [comment]
> ## What does this PR do?
> Fixing variable deployment when creating functions
> 
> ## Test Plan
> [x] Manual testing
> 
> ## Related PRs and Issues
> [appwrite/sdk-for-cli#59](https://github.com/appwrite/sdk-for-cli/issues/59)
> 
> ### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
> Yes

@alibekissakul Please can you add `Closes #59` to the description so the PR gets linked to the issue and after merging it closes the issue and it will also help to know that there is a PR raised for the issue

**Comment by Meldiron** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by stnguyen90** [comment]
@alibekissakul have you had a chance to address all comments? When done, please make sure to re-request a review.

File: templates/cli/lib/commands/deploy.js.twig
Comment (Meldiron) [reply templates/cli/lib/commands/deploy.js.twig:220]: Resolving with my commit.

**Review by Meldiron — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**
